mllib/src/main/scala/org/apache/spark/ml/feature/Bucketizer.scala

```diff
@@ -63,7 +63,7 @@ final class Bucketizer @Since("1.4.0") (@Since("1.4.0") override val uid: String
 
   /** @group setParam */
   @Since("1.4.0")
-  def setSplits(value: Array[Double]): this.type = set(splits, value)
+  def setSplits(value: Array[Double]): this.type = set(splits, value.filter(!_.isNaN))
```

Comment:
How about just rejecting splits with NaN? I don't see a valid reason a caller could specify splits bounded by NaN since it has no ordering.